internal/core/component.rs

```diff
@@ -82,6 +82,16 @@ pub struct ComponentVTable {
     /// strictly speaking not an Item at all!
     pub parent_node: extern "C" fn(core::pin::Pin<VRef<ComponentVTable>>, result: &mut ItemWeak),
 
+    /// Set the parent node of the Component. This Component is part of `parent_component`
+    /// and is included fia a dynamic tree node there at `item_tree_index`.
+    ///
+    /// This uses interior mutability!
```

Comment:
> This uses interior mutability!

What does that mean exactly?  I believe everything in Slint does, so i don't think there is much point mentioning it